qiskit/primitives/containers/observables_array.py

```diff
@@ -98,7 +98,24 @@ def __repr__(self):
         return prefix + array + suffix
 
     def tolist(self) -> list:
-        """Convert to a nested list"""
+        """Convert to a nested list.
+
+        Similar to Numpy's `tolist` method, the level of nesting
+        depends on the dimension of the observables array. In the
+        case of dimension 0 the method returns a single observable
+        instead of a list.
+
+        Examples::
+            Return values for a one-element list vs one element:
+
+                >>> from qiskit.primitives.containers.observables_array import ObservablesArray
+                >>> oa = ObservablesArray.coerce(["Z"])
+                >>> print(type(oa.tolist()))
+                <class 'list'>
+                >>> oa = ObservablesArray.coerce("Z")
+                >>> print(type(oa.tolist()))
+                <class 'dict'>
```

Comment:
Naive question, why is the output type a dict? I would expect a string.

Reply:
In general, an observable can be a weighted sum of Paulis, so it's represented by a dictionary mapping the Pauli (or Pauli string) to its weight. See [here](https://github.com/Qiskit/qiskit/blob/9c70a4df297ff12ce443a9c432e51c1949c4e39c/qiskit/primitives/containers/observables_array.py#L41).